sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -113,6 +113,18 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
       }
     }
 
+    if (requiredSchema.length == 1 &&
+      requiredSchema.head.name == parsedOptions.columnNameOfCorruptRecord) {
+      throw new AnalysisException(
+        s"""
+           |'${parsedOptions.columnNameOfCorruptRecord}' cannot be selected alone without other
+           |data columns. If you want to select corrupt records only, cache or save the Dataset
+           |before executing queries. For example:
```

Comment:
I would explain the downside of this in short, for example, `... as this parses all data under the hood ...`.